General record member referencing (#3516)
modelica/ModelicaSpecification: chapters/arrays.tex

  and does not require that \lstinline!size(a[1].m) = size(a[2].m)!.  The number of subscripts on \lstinline!m! must not be greater than the number of array dimension
  for \lstinline!m! (the number can be smaller, in which case the missing trailing indices are assumed to be `\lstinline!:!'), and is only valid if
  \lstinline!size(a[1].m[$\ldots$]) = size(a[2].m[$\ldots$])!.
\item

>   If \lstinline!...! gives an array of records (e.g. \lstinline!foo(time)!) then \lstinline!(...).m! is interpreted as selecting the \lstinline!m! component in that array of records.

Reviewer: It would be nice to avoid this sort of use of ellipsis.  Maybe like this?
```suggestion
  When the member access operator is applied to a record array, it is interpreted as constructing an array by selecting the member in each record.
```

>   If \lstinline!m! is also an array component, the slice operation is valid only if the resulting array is homogenous.

Reviewer: If the previous sentence doesn't introduce `m`:
```suggestion
  If \lstinline!m! in \lstinline!($\ldots).m!  is also an array component of the record, the slice operation is valid only if the resulting array is homogenous.
```

>   If \lstinline!m! in \lstinline!($\ldots).m!  is also an array component of the record, the slice operation is valid only if the resulting array is homogenous.

Reviewer: Redundant space:
```suggestion
  If \lstinline!m! in \lstinline!($\ldots).m! is also an array component of the record, the slice operation is valid only if the resulting array is homogenous.
```

but it is still possible to perform some operations on them.
\end{example}

\begin{example}

> General selection

Reviewer: There should be some punctuation at the end, and I think a colon is the common choice in a situation like this:
```suggestion
General selection:
```

Reviewer: By the way, we should probably avoid introducing the new terminology _general selection_ in the example:
```suggestion
Member access slicing:
```

\begin{lstlisting}[language=modelica]

>   // Normal slice operation with extra array variable

Reviewer: Avoid making it sound like a poor method:
```suggestion
  // Slice operation as part of component reference:
```

>   Complex c[2]={Complex(1, 2), Complex(2, 3)}*Complex(1, 1);

Reviewer: I find it strange to omit space around the binary operators when there is space after comma:
```suggestion
  Complex c[2] = {Complex(1, 2), Complex(2, 3)} * Complex(1, 1);
```

  Real x1[2]=c.im;

>   // Same result without extra variable
>   // Using generalized member access operator

Reviewer: Emphasize contrast to case above:
```suggestion
  // Same result, but slicing a general expression:
```

>   Real x2[2]=({Complex(1, 2), Complex(2, 3)}*Complex(1, 1)).im;

Reviewer: Similar to above:
```suggestion
  Real x2[2] = ({Complex(1, 2), Complex(2, 3)} * Complex(1, 1)).im;
```
